### PR TITLE
added the k8s network plumbing wg network crd structure

### PR DESCRIPTION
**Pull request by rkamudhan**
see issue #61 and #62 for more info
Complete details in #59

Open source contributors
Co-authored-by: Doug Smith <dougbtv@users.noreply.github.com>
Co-authored-by: Tomofumi Hayashi <s1061123@users.noreply.github.com>
Co-authored-by: Dan Williams <dcbw@users.noreply.github.com>

### CODE REVIEW

File: k8sclient/k8sclient.go [hunk 1]
```diff
@@ -122,34 +122,111 @@ func isJSON(str string) bool {
 	return json.Unmarshal([]byte(str), &js) == nil
 }
 
-func getpluginargs(name string, args string, primary bool, ifname string) (string, error) {
-	var netconf string
-	var tmpargs []string
+func getString(tmpconfig []string) string {
```
[the remark this answers is not in the document]
Comment (rkamudhan) [reply k8sclient/k8sclient.go]: Agreed, I will make the changes

File: k8sclient/k8sclient.go [hunk 2]
```diff
@@ -119,34 +119,101 @@ func parsePodNetworkObject(podnetwork string) ([]map[string]interface{}, error)
...
 	if ifname != "" {
-		tmpargs = append(tmpargs, fmt.Sprintf(`"ifnameRequest": "%s",`, ifname))
+		tmpconfig = append(tmpconfig, fmt.Sprintf(` "ifnameRequest": "%s",`, ifname))
 	}
-	tmpargs = append(tmpargs, args[strings.Index(args, "\""):len(args)-1])
 
-	var str bytes.Buffer
+	tmpconfig = append(tmpconfig, config[1:])
+
+	return strings.Join(tmpconfig,"")
```
Comment (rkamudhan) [inline k8sclient/k8sclient.go]: @s1061123 Added the string join

File: k8sclient/k8sclient.go [hunk 3]
```diff
@@ -119,34 +119,101 @@ func parsePodNetworkObject(podnetwork string) ([]map[string]interface{}, error)
...
+		tmpconfig = append(tmpconfig, `, "masterplugin": true`)
 	}
 
+	if ifname != "" {
+		tmpconfig = append(tmpconfig, fmt.Sprintf(`, "ifnameRequest": "%s"`, ifname))
+	}
+
+	tmpconfig = append(tmpconfig, "}")
+
+	return strings.Join(tmpconfig,"")
```
Comment (rkamudhan) [inline k8sclient/k8sclient.go]: @s1061123 Added the string join

**Comment by rkamudhan** [comment]
@s1061123 @dougbtv All the changes are added in the code. Please review them.

**Comment by s1061123** [comment]
Looks good to, me, too!
thanks, @rkamudhan !

**Pull request merged**